src/server/v2.0/handler/artifact.go

```diff
@@ -16,32 +16,157 @@ package handler
 
 import (
 	"context"
-
+	"fmt"
 	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/goharbor/harbor/src/api/artifact"
+	"github.com/goharbor/harbor/src/pkg/project"
+	"github.com/goharbor/harbor/src/pkg/q"
+	"github.com/goharbor/harbor/src/pkg/repository"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
 	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/artifact"
 )
 
-// ArtifactAPI the api implemention of artifacts
-type ArtifactAPI struct {
+func newArtifactAPI() *artifactAPI {
+	return &artifactAPI{
+		artCtl:  artifact.Ctl,
+		proMgr:  project.Mgr,
+		repoMgr: repository.Mgr,
+	}
+}
+
+type artifactAPI struct {
 	BaseAPI
+	artCtl  artifact.Controller
+	proMgr  project.Manager
+	repoMgr repository.Manager
 }
 
-// DeleteArtifact ...
-func (api *ArtifactAPI) DeleteArtifact(ctx context.Context, params operation.DeleteArtifactParams) middleware.Responder {
-	return operation.NewDeleteArtifactOK()
+// TODO do auth in a separate middleware
+
+func (a *artifactAPI) ListArtifacts(ctx context.Context, params operation.ListArtifactsParams) middleware.Responder {
+	// set query
+	query := &q.Query{
+		Keywords: map[string]interface{}{},
+	}
+	if params.Type != nil {
+		query.Keywords["Type"] = *(params.Type)
+	}
+	if params.Page != nil {
+		query.PageNumber = *(params.Page)
+	}
+	if params.PageSize != nil {
+		query.PageSize = *(params.PageSize)
+	}
+	repository, err := a.repoMgr.GetByName(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName))
```

Comment:
Maybe needed to the controller of respository here.

Reply:
Open issue https://github.com/goharbor/harbor/issues/10552 for that

Reply:
Another option might refer to the repo mgr and project mgr in the artifact controller, not in the handler.s